src/components/PromotedActionsBar.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import {View} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as HeaderUtils from '@libs/HeaderUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import * as Report from '@userActions/Report';
+import * as Session from '@userActions/Session';
+import ROUTES from '@src/ROUTES';
+import type {Report as OnyxReportType} from '@src/types/onyx';
+import Button from './Button';
+import ConfirmModal from './ConfirmModal';
+import type {ThreeDotsMenuItem} from './HeaderWithBackButton/types';
+import * as Expensicons from './Icon/Expensicons';
+
+type PromotedAction = {
+    key: string;
+} & ThreeDotsMenuItem;
+
+type PromotedActionsParams = {
+    report: OnyxReportType;
+};
+
+function usePromotedActions({report}: PromotedActionsParams): Record<string, PromotedAction> {
+    const {translate} = useLocalize();
+    const onShareButtonPress = useCallback(() => Navigation.navigate(ROUTES.REPORT_WITH_ID_DETAILS_SHARE_CODE.getRoute(report?.reportID ?? '')), [report?.reportID]);
+    const join = Session.checkIfActionIsAllowed(() => Report.joinRoom(report));
+
+    return useMemo(
+        () => ({
+            pin: {
+                key: 'pin',
+                ...HeaderUtils.getPinMenuItem(report),
+            },
+            join: {
```

Comment:
Are we going to use `PromotedAction` types `join`, `share`, and `hold` in this PR? If not, let us keep this code commented and add relevant comments as to where this will be used. This can be uncommented later when this is used. This will also keep this PR neat.

Reply:
I've commented this out. However, adding all those actions here, in one place, might be a good idea, as later on, a few PRs might need in simultaneously, leading to conflicts and duplicated work.